### PR TITLE
Upgrading to work with grape 1.4 and fix dependent strategy for mongodb models.

### PR DESCRIPTION
**Pull request by juhguu**

### CODE REVIEW

File: lib/grape_oauth2/responses/base.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def headers
 
         # Raw Rack body
         def raw_body
-          @rack_response[2].body
```
Comment (nbulaj) [inline lib/grape_oauth2/responses/base.rb:43]: Is it a backward compatible change?

File: lib/grape_oauth2/mixins/mongoid/client.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module Client
           include ::Mongoid::Timestamps
 
           has_many :access_tokens, class_name: Grape::OAuth2.config.access_token_class_name,
-                                   foreign_key: :client_id, dependent: :delete
+                                   foreign_key: :client_id, dependent: :delete_all
```
Comment (nbulaj) [inline lib/grape_oauth2/mixins/mongoid/client.rb:14]: Is it a backward compatible change with any versions of Mongoid?

File: lib/grape_oauth2/responses/base.rb (hunk 1 above)
Comment (juhguu) [reply lib/grape_oauth2/responses/base.rb:43]: Hmmm. I haven't tested it. But all tests are passed.

File: lib/grape_oauth2/mixins/mongoid/client.rb (hunk 2 above)
Comment (juhguu) [reply lib/grape_oauth2/mixins/mongoid/client.rb:14]: Mongoid 7.1.

**Pull request closed without merging**